extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallryeGraphqlProcessor.java

```diff
@@ -0,0 +1,376 @@
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.GeneratedResourceBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.LiveReloadBuildItem;
+import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
+import io.quarkus.deployment.configuration.ConfigurationError;
+import io.quarkus.deployment.index.ClassPathArtifactResolver;
+import io.quarkus.deployment.index.ResolvedArtifact;
+import io.quarkus.deployment.util.FileUtil;
+import io.quarkus.deployment.util.ServiceUtil;
+import io.quarkus.runtime.LaunchMode;
+import io.quarkus.smallrye.graphql.runtime.SmallRyeGraphQLRecorder;
+import io.quarkus.vertx.http.deployment.HttpRootPathBuildItem;
+import io.quarkus.vertx.http.deployment.RequireBodyHandlerBuildItem;
+import io.quarkus.vertx.http.deployment.RouteBuildItem;
+import io.quarkus.vertx.http.deployment.devmode.NotFoundPageDisplayableEndpointBuildItem;
+import io.quarkus.vertx.http.runtime.HandlerType;
+import io.smallrye.graphql.cdi.config.ConfigKey;
+import io.smallrye.graphql.cdi.config.GraphQLConfig;
+import io.smallrye.graphql.cdi.producer.GraphQLProducer;
+import io.smallrye.graphql.schema.Annotations;
+import io.smallrye.graphql.schema.SchemaBuilder;
+import io.smallrye.graphql.schema.model.Operation;
+import io.smallrye.graphql.schema.model.Reference;
+import io.smallrye.graphql.schema.model.Schema;
+import io.smallrye.graphql.spi.LookupService;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Processor for Smallrye GraphQL.
+ * We scan all annotations and build the model during build.
+ */
+public class SmallryeGraphqlProcessor {
+
+    private static final Logger LOG = Logger.getLogger(SmallryeGraphqlProcessor.class.getName());
+    private static final String SCHEMA_PATH = "/schema.graphql";
+    private static final String SPI_PATH = "META-INF/services/";
+
+    // For the UI
+    private static final String GRAPHQL_UI_WEBJAR_GROUP_ID = "io.smallrye";
+    private static final String GRAPHQL_UI_WEBJAR_ARTIFACT_ID = "smallrye-graphql-ui-graphiql";
+    private static final String GRAPHQL_UI_WEBJAR_PREFIX = "META-INF/resources/graphql-ui";
+    private static final String OWN_MEDIA_FOLDER = "META-INF/resources/";
+    private static final String GRAPHQL_UI_FINAL_DESTINATION = "META-INF/graphql-ui-files";
+    private static final String TEMP_DIR_PREFIX = "quarkus-graphql-ui_" + System.nanoTime();
+    private static final List<String> IGNORE_LIST = Arrays.asList(new String[] { "logo.png", "favicon.ico" });
+    private static final String FILE_TO_UPDATE = "render.js";
+
+    SmallRyeGraphQLConfig quarkusConfig;
```

Comment:
I am getting a compile error now, seems you forgot to add this class after extracting it

Reply:
Sorry. (But it build on my machine :))
I'll commit now with the other Arc fix

Reply:
@jmartisk please check now